src/dfx/src/commands/replica.rs

```diff
@@ -175,6 +185,21 @@ pub fn exec(
         artificial_delay,
     )?;
 
+    let effective_config = if emulator {
+        CachedConfig::emulator()
+    } else {
+        CachedConfig::replica(&replica_config)
+    };
+    if !force && previous_config_path.exists() {
+        let previous_config = load_json_file(&previous_config_path)
+            .context("Failed to read replica configuration. Run `dfx start` with `--clean`.")?;
+        if effective_config != previous_config {
+            bail!("The network configuration was changed. Run `dfx start` with `--clean`.")
+        }
+    }
+    save_json_file(&previous_config_path, &effective_config)
+        .context("Failed to write replica configuration")?;
+
```

Comment:
I missed something: this can't be moved here easily for the time being, because `replica_config` is altered below on lines 239-250 immediately before the call to `start_replica_actor`.